include/rpt/particle_detector_interactions.h

```diff
@@ -0,0 +1,186 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2020 - by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 3.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+
+*
+* Authors: Audrey Collard-Daigneault, Polytechnique Montreal, 2021-
+*/
+
+#ifndef lethe_particle_detector_interactions_h
+#define lethe_particle_detector_interactions_h
+
+/**
+ * This class allows to calculate the photon count from a particle received by
+ * a detector with the Monte Carlo method.
+ */
+
+#include <deal.II/base/point.h>
+#include <deal.II/base/tensor.h>
+
+#include <rpt/detector.h>
+#include <rpt/parameters_rpt.h>
+#include <rpt/radioactive_particle.h>
+#include <rpt/rpt_calculating_parameters.h>
+
+template <int dim>
+class ParticleDetectorInteractions
+{
+public:
+  /**
+   * @brief Constructor for the ParticleDetectorInteractions.
+   *
+   * @param particle Particle which contains information about its position
+   *
+   * @param detector Detector which contains information about its positions,
+   * radius and length
+   *
+   * @param rpt_parameters All other parameters needed for the count calculation
+   *
+   */
+  ParticleDetectorInteractions(RadioParticle<dim> &      particle,
+                               Detector<dim> &           detector,
+                               RPTCalculatingParameters &rpt_parameters)
+    : particle_position(particle.get_position())
+    , detector_face_position(detector.get_face_position())
+    , detector_middle_position(detector.get_middle_position())
+    , detector_radius(detector.get_radius())
+    , detector_length(detector.get_length())
+    , fixed_parameters(rpt_parameters.rpt_param)
+    , initial_parameters(rpt_parameters.initial_param)
+  {}
+
+  /**
+   * @brief Calculate photon count of a detector with the Monte Carlo method.
+   */
+  double
+  calculate_count();
+
+  double
+  get_h();
+
+  double
+  get_rho();
+
+  double
+  get_alpha(double n_alpha, double n_theta);
+
+  double
+  get_theta(double n_alpha, double n_theta);
```

Comment:
is this a getter or do these function calculate theta and alpha? In the latter case, they should not be named getters :)

Reply:
Actually, those getter are for the unit tests because my function calculate_solid_angle() is private. I think I can just have "return alpha" but to have the angles (and not empty doubles) but calculate_position_parameters() & calculate_solid_angle have to be called. I can do it by hand in unit tests but I wanted to avoid someone use it but get issues because the methods were not call before. 
Should I keep it as is, change to only "return alpha/theta" or change my methods to public?

Reply:
Nope that's good. I was just wondering if it was a getter or not.
You can maybe add a comment above those to say : "Getter used for unit testing purposes"
It's a good design what you did :).